lib/sparqlAlgebra.ts

```diff
@@ -301,8 +301,18 @@ function translateGroupGraphPattern(thingy: Pattern) : Algebra.Operation
 
 function translateExpression(exp: Expression | RDF.Term | Wildcard) : Algebra.Expression
 {
-    if (Util.isSimpleTerm(exp) || Util.isQuad(exp))
+    if (Util.isSimpleTerm(exp))
         return factory.createTermExpression(exp);
+    if (Util.isQuad(exp)) {
+        if (Util.hasQuadVariables(exp))
+            return factory.createOperatorExpression('triple', [
+              translateExpression(exp.subject),
+              translateExpression(exp.predicate),
+              translateExpression(exp.object),
+            ]);
+        else
+            return factory.createTermExpression(exp);
+    }
```

Comment:
What is the reason this is necessary? From what I can tell this replaces the triple with a function that would return the same triple. Or am I missing some weird constructions that can occur now?

Reply:
This is needed because `<< ?s ?p ?o >>` is illegal in an expression due to the variables, that's why it needs to be converted to the dedicated expression function `TRIPLE(?s ?p ?o)`.